zebra-chain/src/orchard/tree.rs

```diff
@@ -99,7 +97,7 @@ lazy_static! {
 /// The root hash in LEBS2OSP256(rt) encoding of the Orchard note commitment
 /// tree corresponding to the final Orchard treestate of this block. A root of a
 /// note commitment tree is associated with each treestate.
-#[derive(Clone, Copy, Default, Eq, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Copy, Default, Eq, Serialize, Deserialize)]
```

Comment:
I wasn't able to reproduce the lint warning on nightly, are you sure we can't derive `PartialEq` here?

Reply:
The problem is that the lint is renamed, so we have to use either the old or new name. But we also want to compile without warnings on stable and nightly.

The easiest solution is to fix the underlying problem the lint is identifying.